### PR TITLE
Update PULL_REQUEST_TEMPLATE with 2.0 release notes section

### PR DESCRIPTION
**Pull request by marcotc**
This PR ensures that developers are prompted with describing how 2.0 changes will affect users, and how users can migrate their applications successfully to 2.0.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -5,6 +5,17 @@ for guidance on how to set up your development environment,
 run the test suite, write new integrations, and more.
 -->
 
+**2.0 Upgrade Guide notes**
+<!--
+(If this PR is for 1.x, please delete this section)
+A description that will into the https://github.com/DataDog/dd-trace-rb/blob/master/docs/UpgradeGuide.md for this change.
```
Comment (ekump) [inline .github/PULL_REQUEST_TEMPLATE.md:11]:
```suggestion
A public facing description that will go into the [upgrade guide](https://github.com/DataDog/dd-trace-rb/blob/master/docs/UpgradeGuide.md) for this change.
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -5,6 +5,17 @@ for guidance on how to set up your development environment,
...
+We already know what the PR does (from the PR title),
+but users should know either:
+* A migration path for this change. In other words, how to they continue using their application without behavior changes
```
Comment (ekump) [inline .github/PULL_REQUEST_TEMPLATE.md:14]:
```suggestion
* A migration path for this change. In other words, how to continue using their application without behavior changes
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -5,6 +5,17 @@ for guidance on how to set up your development environment,
 run the test suite, write new integrations, and more.
 -->
 
+**2.0 Upgrade Guide notes**
```
Comment (ekump) [inline .github/PULL_REQUEST_TEMPLATE.md:8]: Should this go towards the bottom under the heading "For Datadog Employees"?

**Review by ekump — APPROVED** [review]
Made a couple of suggestions, but LGTM

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (marcotc) [reply .github/PULL_REQUEST_TEMPLATE.md:8]: I think it should stay on top for visibility.
I think there's a good chance we might miss it if we move it down.

**Review by ekump — APPROVED** [review]

**Pull request merged**